### DISCUSSION TITLE
New @supabase/supabase-js version giving IDE errors

### DISCUSSION THREAD

**Discussion by mambucodev** [supabase/supabase#30169] [Questions] [answered]
Now I don't know if I'm doing anything wrong here, but a fully functional code is getting IDE errors (and I repeat, this code works):

![image](https://github.com/user-attachments/assets/5d27395d-a577-4495-8923-6a446b40da7a)

```ts
supabase
        .from('posts')
        .select(
          `
        id,
        name,
        path,
        description,
        date,
        banner,
        sections ( name, display_name )
      `,
          { count: 'estimated' }
        )
        .order('id', { ascending: false })
        .range(page * 9 - 9, page * 9 - 1)
        .then(({ data, count, error }) => {
          if (error) {
            setError(error);
            console.error(error);
          } else {
            setPosts(data);
            setMaxPage(Math.ceil(count! / 9));
```

**Comment by mambucodev** [comment]
Actually, I'd focus on this:
![image](https://github.com/user-attachments/assets/9e93d939-0f0d-4940-afde-881cb89837dc)

**Reply by GaryAustin1** [reply]
Please generate a support request.

**Reply by GaryAustin1** [reply]
Another user was able to get going by reverting supabase-js back on version 2.46 to 2.45.6.

**Reply by mambucodev** [reply]
Well, this isn't a problem for me, I added a @ts-expect-error so everything is chill. the thing is, there's this error related to how they defined Types.

Nothing to worry about, next.js linter even ignores it.

**Reply by f0n2** [reply]
> Another user was able to get going by reverting supabase-js back on version 2.46 to 2.45.6.

I am having the same issue. All my fetching stopped working due to this error. Reverting versions fixed it.

**Reply by mambucodev** [reply, marked as answer]
I'm letting everyone know that @supabase/supabase-js just released a new version and everything got fixed. For who's still on the older version, the temporary fix was adding:
```ts
// @ts-expect-error <your reason>
```
just before every supabase-related error.
